hc2/headers/functions/hsa_interfaces.hpp

```diff
@@ -0,0 +1,299 @@
+//===----------------------------------------------------------------------===//
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include "../types/raii_handle.hpp"
+
+#include <hc.hpp>
+
+#include <hsa/hsa.h>
+
+#include <algorithm>
+#include <cstddef>
+#include <functional>
+#include <memory>
+#include <utility>
+
+inline
+bool operator==(const hsa_agent_t& x, const hsa_agent_t& y)
```

Comment:
can we add "static" to these free functions?

Reply:
@scchan we could but in this context I think it may not be very useful. Are you concerned about potential ODR violations if someone else decides to make hsa_agent_t EqualityComparable? The downside is that now you technically may get an operator== code drop in all TUs which include that header (an academic issue, agreed). I do not feel strongly either way, and these are temporary...

Reply:
It's mainly about potential ODR violations.  We have other headers in hcc containing inlined functions  without `static` and they are included in multiple .cpp files.  That is causing some portability issues with some compilers.  We are looking into fixing them.

Reply:
@scchan done:)